File: captcha-frontend/captcha-verification/src/components/testing_methods/python testing methods/stealth_bot-1/behaviors/scroll.py
import time
import random

def scroll(driver):
    print("🌀 Scrolling...")
    for _ in range(random.randint(5, 10)):
        driver.execute_script("window.scrollBy(0, arguments[0]);", random.randint(100, 300))
        time.sleep(random.uniform(0.3, 0.6))

File: captcha-frontend/captcha-verification/src/components/testing_methods/JS_testing_methods/botsimulator/behaviors/scroll.js
const { randomInt, sleep } = require('../utils/random');

async function scroll(page) {
  try {
    const isScrollable = await page.evaluate(() => {
      return document.body.scrollHeight > window.innerHeight;
    });

    if (!isScrollable) {
      console.warn('⚠️ Page is not scrollable. Skipping scroll behavior.');
      return;
    }

    console.log('🌀 Scrolling the page...');
    for (let i = 0; i < randomInt(5, 15); i++) {
      const step = randomInt(100, 500);
      await page.evaluate((step) => {
        window.scrollBy({ top: step, behavior: 'smooth' });
      }, step);
      await sleep(randomInt(400, 800));
    }

    console.log('✅ Done scrolling.');
  } catch (err) {
    console.error('❌ Scroll behavior failed:', err.message);
  }
}

module.exports = scroll;
